components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultRequestCoordinator.java

```diff
@@ -613,7 +613,9 @@ protected void findPreviousAuthenticatedSession(HttpServletRequest request, Auth
             String sessionContextKey = DigestUtils.sha256Hex(cookie.getValue());
 
             // get the authentication details from the cache
+            FrameworkUtils.startTenantFlow(context.getTenantDomain());
```

Comment:
Shouldn't we do the tenantFlow start and end within a try-finally{} block

Reply:
FrameworkUtils.getSessionContextFromCache(sessionContextKey) doesn't throw any exception. Hence, shall we proceed as it is?